HotSpot/View Controllers/EditInformationViewController.m

```diff
@@ -0,0 +1,116 @@
+@property (strong, nonatomic) PFUser *currentUser;
+@property (strong, nonatomic) UIAlertController *alert;
+
+@end
+
+@implementation EditInformationViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.currentUser = [PFUser currentUser];
+    [self.currentUser saveInBackground];
+    self.fullName.text = self.currentUser[@"name"];
+    self.phoneNumber.text = self.currentUser[@"phone"];
+    self.email.text = self.currentUser[@"email"];
+    self.username.text = self.currentUser.username;
+    
+    self.alert = [UIAlertController alertControllerWithTitle:@""
+                                                                   message:@"" preferredStyle:UIAlertControllerStyleAlert];
+    
+    // create an OK action
+    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK"
+                                                       style:UIAlertActionStyleDefault
+                                                     handler:^(UIAlertAction * _Nonnull action) {
+                                                         nil;
+                                                     }];
+    // add the OK action to the alert controller
+    [self.alert addAction:okAction];
+}
+
+- (IBAction)didTapSaveChanges:(UIButton *)sender {
+    if (![self.currentUser[@"name"] isEqualToString:(self.fullName.text)]) {
+        self.currentUser[@"name"] = self.fullName.text;
+    }
+    if (![self.currentUser.username isEqualToString:(self.username.text)]) {
+        self.currentUser.username = self.username.text;
+    }
+    if (![self.currentUser[@"phone"] isEqualToString:(self.phoneNumber.text)]) {
+        self.currentUser[@"phone"] = self.phoneNumber.text;
+    }
+    if (![self.currentUser[@"email"] isEqualToString:(self.email.text)]) {
+        self.currentUser[@"email"] = self.email.text;
+    }
+    [self.currentUser saveInBackground];
+    [self dismissViewControllerAnimated:YES completion:nil];
+}
+
+- (IBAction)didEditFullName:(UITextField *)sender {
+    if (self.fullName.text.length == 0) {
+        self.alert.title = @"Full Name Change Error";
+        self.alert.message = @"Your full name cannot be empty";
+        [self presentViewController:self.alert animated:YES completion:^{
+        }];
+    }
+}
+
+- (IBAction)didEditUsername:(UITextField *)sender {
+    if (self.username.text.length == 0) {
+        self.alert.title = @"Username Change Error";
+        self.alert.message = @"Your username cannot be empty";
+        [self presentViewController:self.alert animated:YES completion:^{
+        }];
+    }
+}
+
+- (IBAction)didEditEmail:(UITextField *)sender {
+    if (self.email.text.length == 0) {
+        self.alert.title = @"Email Change Error";
+        self.alert.message = @"Your email cannot be empty";
+        [self presentViewController:self.alert animated:YES completion:^{
+        }];
+    }
+}
+
+- (IBAction)didEditPhoneNumber:(UITextField *)sender {
+    if (self.phoneNumber.text.length != 10) {
```

Comment:
might want to add regex for email, phone number, etc. to validate them. for example, phone numbers should be 10 characters and all numbers. emails need to have the format something@domain.com